src/lib.rs

```diff
@@ -690,7 +690,9 @@ where
                 } else {
                     // We're keeping this bin so decrement the quantity of the available stock
                     // piece.
-                    stock_piece.dec_quantity();
+                    if stock_piece.quantity != Some(0) {
```

Comment:
You're correct that we need to check that the quantity isn't 0, but this check should be in the `find` on line 678. It's supposed to find an available stock piece, but I forgot to make sure it's a positive quantity. So before checking if the `bin` matches `sp`, we need to make sure the quantity is positive.

Reply:
All done...